pkg/utils/kubernetes/backupinfrastructure.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	garden "github.com/gardener/gardener/pkg/client/garden/clientset/versioned"
+	"github.com/gardener/gardener/pkg/logger"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+)
+
+func tryBackupInfrastructure(
```

Comment:
too much nesting, I would suggest adding documentation to this function

Reply:
That's an internal function (similar to what @adracus has implement the first time for shoots: https://github.com/gardener/gardener/blob/master/pkg/utils/kubernetes/shoot.go#L27) --> no docs needed? What should be documented there?.
What do you mean with "too much nesting"? I only see one level?

Reply:
some internal functions are also documented in Kubernetes, have a look 
[here](https://github.com/gardener/gardener/blob/master/vendor/k8s.io/apimachinery/pkg/runtime/scheme.go#L114) for an example. Furthermore, we can also document @adracus function if needed. Maybe nesting here is a bad word, I meant too much First-class funcs without documentation can confuse moral humans.

Reply:
This should be `TryUpdateBackupInfrastructure`.
Regarding the comments, I am a bit split - I would also be fine without them and just for the exported functions. With #583 we anyways can factor this out and write it once for all types, guess there we can make stuff even better.